src/main/scala/zalando/meetup/bracket/DatabaseOperations.scala

```diff
@@ -25,10 +24,20 @@ object DatabaseOperations {
   def createDataSource[F[_] : Sync](conf: DatabaseConfig): Resource[F, HikariDataSource] =
     Resource.fromAutoCloseable(createDSConfig[F](conf).map(new HikariDataSource(_)))
 
+  def connectionResource[F[_]: Sync](ds: DataSource): Resource[F, Connection] = Resource.fromAutoCloseable(Sync[F].catchNonFatal(ds.getConnection()))
+
+  private def findCustomerStmt[F[_]: Sync](connection: Connection, id: Long): F[PreparedStatement] = { 
+    Sync[F].delay { 
+      val stmt = connection.prepareStatement("SELECT name from CUSTOMERS WHERE customer_id = ?") 
+      stmt.setLong(1, id)
```

Comment:
Hmm, this is not what should be done, as if this fails we won't return the statement that Resource needs to cleanup. That is why it was that way

Reply:
setLong can only fail if it's a bug in the system. I don't think it should be catched... but detected and corrected.

Reply:
Not sure about that, who knows what can happen in there, also the goal was to demonstrate how to do this when the operations are impure

Reply:
Ok, can do...

Reply:
Thanks